### PR TITLE
Implicitly verify biallelic.

### PR DESCRIPTION
**Pull request by cseed**
Implicitly verify biallelic on non-split VDSes on the python side for methods that require it through the @require_biallelic decorator.
    
Scala interface now `require`s biallelic expectations.  Other requirements should be written in this form, too (like TSampleString).  This is based on the decision not to support a user-level Scala interface that mirror the Python.  We've already left that with, say, history.

### CODE REVIEW

**Comment by cseed** [comment]
Builds on: https://github.com/hail-is/hail/pull/2218

File: src/main/scala/is/hail/annotations/MemoryBlock.scala [hunk 1]
```diff
@@ -13,7 +17,7 @@ object MemoryBuffer {
   }
 }
 
-final class MemoryBuffer(var mem: Long, var length: Long, var offset: Long = 0) {
+final class MemoryBuffer(var mem: Long, var length: Long, var offset: Long = 0) extends KryoSerializable with Serializable {
```
[the remark this answers is not in the document]
Comment (cseed) [reply src/main/scala/is/hail/annotations/MemoryBlock.scala]: I had built on another PR, but I rebase it directly onto master.

File: src/main/scala/is/hail/io/vcf/LoadVCF.scala [hunk 2]
```diff
@@ -329,8 +322,7 @@ object LoadVCF {
       TVariant(gr),
       vaSignature,
       TStruct.empty,
-      genotypeSignature,
-      wasSplit = noMulti),
```
[the remark this answers is not in the document]
Comment (cseed) [reply src/main/scala/is/hail/io/vcf/LoadVCF.scala]: I don't understand what there is to do for LoadBgen.  Can you be more specific?  It doesn't do the double-scan like LoadVCF.  (I see you have a separate, related PR.  I will take a look shortly.)

File: src/main/scala/is/hail/variant/VariantDataset.scala [hunk 3]
```diff
@@ -107,7 +102,8 @@ class VariantDatasetFunctions(private val vds: VariantDataset) extends AnyVal {
   }
 
   def concordance(other: VariantDataset): (IndexedSeq[IndexedSeq[Long]], KeyTable, KeyTable) = {
-    requireSplit("concordance")
+    require(vds.wasSplit)
+    require(other.wasSplit)
 
     if (!other.wasSplit)
```
[the remark this answers is not in the document]
Comment (cseed) [reply src/main/scala/is/hail/variant/VariantDataset.scala]: Good catch!

**Comment by cseed** [comment]
Addressed comments.  Can you take another look @jigold?

File: src/main/scala/is/hail/io/vcf/LoadVCF.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (jigold) [reply src/main/scala/is/hail/io/vcf/LoadVCF.scala]: I think you need to delete the `wasSplit = true` line in BgenLoader. Line 85.

[the remark this answers is not in the document]
Comment (cseed) [reply src/main/scala/is/hail/io/vcf/LoadVCF.scala]: I understand.  Got it.

**Pull request merged**